llrt_core/src/modules/http/request.rs

```diff
@@ -207,18 +220,33 @@ fn assign_request<'js>(request: &mut Request<'js>, ctx: Ctx<'js>, obj: &Object<'
                 ));
             }
 
-            request.body = if let Some(blob) = get_class::<Blob>(&body)? {
-                let blob = blob.borrow();
-                Some(TypedArray::<u8>::new(ctx.clone(), blob.get_bytes())?.into_value())
-            } else {
-                Some(body)
+            match get_class::<Blob>(&body)? {
+                Some(blob) => {
+                    let blob = blob.borrow();
+                    request.body =
+                        Some(TypedArray::<u8>::new(ctx.clone(), blob.get_bytes())?.into_value());
+                    request.content_type = match blob.mime_type().len() {
+                        0 => None,
+                        _ => Some(blob.mime_type()),
+                    };
+                },
+                None => {
+                    request.body = Some(body);
+                    request.content_type = None;
+                },
             }
         }
     }
 
     if obj.contains_key("headers").unwrap() {
         let headers: Value = obj.get("headers")?;
         let headers = Headers::from_value(&ctx, headers)?;
+        if request.content_type.is_none() {
```

Comment:
This is unnecessary since we only need to set `request.content_type` if blob is being used as body or if `blob()` method is called. It's thus better to calculate this from the blob method only. In this implementation we'll loop through all headers even if we don't ever use `blob()` method.

Reply:
I have decided to refer to the content-type of Headers directly in Blob(), so this support is no longer necessary.